connectors/byoei.py

```diff
@@ -409,17 +444,61 @@ async def get_existing_ids(self, index):
             ts = doc["_source"].get(TIMESTAMP_FIELD)
             yield doc_id, ts
 
+    def done(self):
+        if self._fetcher_task is not None and not self._fetcher_task.done():
```

Comment:
you can reunite everything in a single op:

```python
return (
  (self._fetcher_task is None and self._bulker_task is None) or    # no tasks
  (self._fetcher_task.done() and self._bulker_task.done())         
) 
```

Reply:
If one of `_fetcher_task` and `_bulker_task` is None, this will fail.